pkg/cmd/agent/list.go

```diff
@@ -23,30 +24,42 @@ func NewCmdAgentList(f *factory.Factory) *cobra.Command {
 			List all connected agents for the current organization.
 		`),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			opts := buildkite.AgentListOptions{
-				Name:     name,
-				Hostname: hostname,
-				Version:  version,
-			}
-			loader := func() tea.Msg {
-				agents, _, err := f.RestAPIClient.Agents.List(f.Config.Organization, &opts)
-				items := make(agent.NewAgentItemsMsg, len(agents))
+			loader := func(page, perpage int) tea.Cmd {
+				return func() tea.Msg {
+					opts := buildkite.AgentListOptions{
+						Name:     name,
+						Hostname: hostname,
+						Version:  version,
+						ListOptions: buildkite.ListOptions{
+							Page:    page,
+							PerPage: perpage,
+						},
+					}
 
-				if err != nil {
-					return err
-				}
+					agents, resp, err := f.RestAPIClient.Agents.List(f.Config.Organization, &opts)
+					items := make([]agent.AgentListItem, len(agents))
+
+					if err != nil {
+						return err
+					}
+
+					for i, a := range agents {
+						a := a
+						items[i] = agent.AgentListItem{
+							Agent: &a,
+						}
+					}
 
-				for i, a := range agents {
-					items[i] = agent.AgentListItem{
-						Agent: &a,
+					return agent.NewAgentItemsMsg{
+						Items:    items,
+						LastPage: resp.LastPage,
 					}
 				}
-				return items
 			}
 
-			model := agent.NewAgentList(loader)
+			model := agent.NewAgentList(loader, 1, perpage)
 
-			p := tea.NewProgram(model, tea.WithAltScreen())
```

Comment:
We still want to use the alternate screen here

Reply:
Case for me removing it on tests - have brought it back with a fresh commit